### ISSUE TITLE
Push via HTTP with credentials on Windows

### ISSUE THREAD

**Issue by kylekatarnls** [jacogr/atom-git-control#61]
Hi,

I try to push my repo with the "Push" button (enabled) but nothing happen. It is a HTTP remote and it need credentials. The following command appears but nothing else:
```git -c push.default=simple push --porcelain```
If I type the same in command-line, it ask me my credentials and it works.

I'm on Windows.

**Comment by MarcelMue** [comment]
Hi @kylekatarnls , 
This seems to be the same issue as #7 .
So far our only guess is that the environment inside our [shelljs](https://github.com/arturadib/shelljs) shell used by [git-promise](https://github.com/piuccio/git-promise) doesn't get propagated properly. 
I currently don't have the time to look into this issue too much as it would (most likely) need significant changes to our current setup to fix this problem. It is on our roadmap however.

Contributions or Pull requests are welcome though :D

**Comment by kylekatarnls** [comment]
I got the same conclusion and was trying to propose a pull request to shelljs to handle the command read-line. Then a pull request will be needed at git-promise to handle credentials asking, then a form to display it wil be able to be done in atom-git-control. Sound like it wil take some time but I will try to do it, this feature would be very nice to me.

**Comment by MarcelMue** [comment]
I am not trying to be a downer but a pull request getting merged into shelljs does not seem very likely to me unfortunately. The repo has 23 open pull requests right now and in all of 2015 exactly one outside pull request has been merged.

At this point I think it would be a better idea to try and look for other options from shelljs.Maybe Atoms [Buffered Process](https://atom.io/docs/api/v0.194.0/BufferedProcess) could do the job, but I am honestly not experienced enough with Node to tell :(

**Comment by kylekatarnls** [comment]
https://github.com/jprichardson/node-suppose handle it bu is no more up to date.

**Comment by bnb** [comment]
Also having this problem. Would really like to see a fix for this.

**Comment by hellerbarde** [comment]
[See this for a properly set up credential storage on Windows/Linux](https://github.com/jacogr/atom-git-control/issues/42#issuecomment-121604138)
Hope that helps. :+1:

**Comment by vnabet** [comment]
For credentials you can use [Git Crendetial Manager for Windows](https://github.com/Microsoft/Git-Credential-Manager-for-Windows)

Make, for instance, a pull in a command line to register your credentials. Then Push command will work fine with atom git-control

**Comment by kylekatarnls** [comment]
Yes but it is not a way as secured as asking for each push the password.

**Comment by hellerbarde** [comment]
@kylekatarnls What do you propose as an alternative? Entering passwords all the time is not a good solution either.

**Comment by kylekatarnls** [comment]
Why not, if you have a password or a key with a passphrase, the most secure way to use it is to enter it  each time, or each session. In fact, the most agnostic way to deal with it would be to ask the user with a prompt box for example for any input from the console. It is possible to use wincreds storage (for example with a checkbox). But IMHO, atom-git-control should let the choice to the user as git do.
